pxc/pxc.go

```diff
@@ -225,57 +224,18 @@ func (p *PXC) SubtractGTIDSet(ctx context.Context, set, subSet string) (string,
 	return result, nil
 }
 
-func getNodesByServiceName(ctx context.Context, pxcServiceName string) ([]string, error) {
-	cmd := exec.CommandContext(ctx, "/opt/percona/peer-list", "-on-start=/usr/bin/get-pxc-state", "-service="+pxcServiceName)
-	out, err := cmd.CombinedOutput()
-	if err != nil {
-		return nil, errors.Wrap(err, "get peer-list output")
-	}
-	return strings.Split(string(out), "node:"), nil
-}
-
-func GetPXCFirstHost(ctx context.Context, pxcServiceName string) (string, error) {
-	nodes, err := getNodesByServiceName(ctx, pxcServiceName)
-	if err != nil {
-		return "", errors.Wrap(err, "get nodes by service name")
-	}
-	sort.Strings(nodes)
-	lastHost := ""
-	for _, node := range nodes {
-		if strings.Contains(node, "wsrep_ready:ON:wsrep_connected:ON:wsrep_local_state_comment:Synced:wsrep_cluster_status:Primary") {
-			nodeArr := strings.Split(node, ":")
-			lastHost = nodeArr[0]
-			break
-		}
-	}
-	if len(lastHost) == 0 {
-		return "", errors.New("can't find host")
-	}
-
-	return lastHost, nil
-}
-
-func GetPXCOldestBinlogHost(ctx context.Context, pxcServiceName, user, pass string) (string, error) {
-	nodes, err := getNodesByServiceName(ctx, pxcServiceName)
-	if err != nil {
-		return "", errors.Wrap(err, "get nodes by service name")
-	}
-
+func GetPXCOldestBinlogHost(ctx context.Context, hosts []string, user, pass string) (string, error) {
 	var oldestHost string
 	var oldestTS int64
-	for _, node := range nodes {
-		if strings.Contains(node, "wsrep_ready:ON:wsrep_connected:ON:wsrep_local_state_comment:Synced:wsrep_cluster_status:Primary") {
```

Comment:
this line checks if member is Online and Primary.
it is possible to get similar info from `SELECT MEMBER_HOST,MEMBER_STATE,MEMBER_ROLE FROM performance_schema.replication_group_members;`

Reply:
Do you mean that mysql-pitr-helper must receive only the single host and then deduces every other host with this?

Reply:
Also, it can be possible to cycle-through provided hosts, run your command on the first one available and then use hosts from the output. Double work though.
Also, as I understand, wsrep_cluster_status:Primary refers to the cluster's state, not the unit's one, so `MEMBER_ROLE` isn't right alternative for that, am I right?

Reply:
Helper can receive all hosts but we must to check member health each time when we connect.
The simplest way to check if member is still good is to query replication_group_members